### ISSUE TITLE
Scalebar is missing in pdfs generated from pyplot.savefig() 

### ISSUE THREAD

**Issue by ltizei** [hyperspy/hyperspy#2374]
For figures generated from s.plot(), when saving to .pdf (only) two things are systematically not included in the pdf: 1) the scale bar and 2) the scale of the colorbar (but the colorbar is fine). Saves to .png work fine.
The same thing does not happen when using the notebook as output.
This was observed in:
1) Python 3.8.2, Matplolib 3.1.3, Hyperspy 1.5.2 and Qt 5.9.7 on Ubuntu 18.04.4;
2) on matplotlib 3.1.1, python 3.7.6, qt 5.9.7 and the non_uniform_axes branch of hyperspy on windows 10;

Example: 
```
%matplotlib qt
import hyperspy.api as hs
import matplotlib.pyplot as plt
s = hs.datasets.example_signals.object_hologram()
s.plot()
plt.savefig('Trial.pdf')
```

The following workaround works:

```
from matplotlib.backends.backend_pdf import PdfPages
s = hs.datasets.example_signals.object_hologram()
s.plot()
with PdfPages('Trial.pdf') as pdf:
     pdf.savefig()
```

**Comment by ericpre** [comment]
Since the issue does occur when using the notebook backend (which doesn't support blitting), I suspect this is an issue with the animated artists (which are animated only with backend supporting blitting) not being drawn with the pdf backend. We need a minimum example using matplotlib only to narrow down the issue.
@ltizei: you mention on gitter that it worked in the past, do you know a combination of version of matplotlib and hyperspy, which is working fine?

**Comment by ericpre** [comment]
Labelled as bug with a question because this is not clear if the issue is in hyperspy or matplotlib.

**Comment by ltizei** [comment]
The change happened a bit more than 6 months ago, if my memory does not fail me. I will try some older versions to pin it down.

**Comment by ltizei** [comment]
I tried older versions (list below). The problem seems to have shown up after the update from matplotlib 2.2.4 to 3.0.3. Hyperspy 1.5.2 does print scale bars in pdfs with matplotlib 2.2.4.

scale bar in pdf ok:
python 3.6.10, hyperspy = 1.3.1,matplotlib=2.2.4  
python 3.6.10, hyperspy = 1.5.2,matplotlib=2.2.4 
python 3.8.2, hyperspy = 1.5.2,matplotlib=2.2.4 


Scale bar not ok in pdf:

python 3.8.2, hyperspy = 1.5.2, matplotlib=3.2.1  
python 3.8.2, hyperspy = 1.5, matplotlib=3.2.1   
python 3.6.10, hyperspy = 1.4.1, matplotlib=3.0.3
python 3.6.10, hyperspy = 1.3.1,matplotlib=3.0.3

**Comment by ericpre** [comment]
https://github.com/matplotlib/matplotlib/issues/17599

**Comment by ericpre** [comment]
@ltizei, @LMSC-NTappy, this is fixed in the matplotlib master branch and should be included in the coming matplotlib v3.2.2.
